src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Generators/IVsSingleFileGeneratorFactoryAggregator.cs

```diff
@@ -0,0 +1,104 @@
+        {
+            Requires.NotNull(serviceProvider, nameof(serviceProvider));
+            Requires.NotNull(registryHelper, nameof(registryHelper));
+            _serviceProvider = serviceProvider;
+            _registryHelper = registryHelper;
+        }
+
+        public int CreateGeneratorInstance(string wszProgId, out int pbGeneratesDesignTimeSource, out int pbGeneratesSharedDesignTimeSource, out int pbUseTempPEFlag, out IVsSingleFileGenerator ppGenerate)
+        {
+            // The only user in the project system does not call this method.
+            throw new NotImplementedException();
+        }
+
+        public int GetDefaultGenerator(string wszFilename, out string pbstrGenProgID)
+        {
+            // The only user in the project system does not call this method.
+            throw new NotImplementedException();
+        }
+
+        public int GetGeneratorInformation(string wszProgId, out int pbGeneratesDesignTimeSource, out int pbGeneratesSharedDesignTimeSource, out int pbUseTempPEFlag, out Guid pguidGenerator)
+        {
+            Requires.NotNullOrEmpty(wszProgId, nameof(wszProgId));
+            EnsureInitialized(true);
+
+            pbGeneratesDesignTimeSource = 0;
+            pbGeneratesSharedDesignTimeSource = 0;
+            pbUseTempPEFlag = 0;
+            pguidGenerator = Guid.Empty;
+
+            var generatorKey = GetGeneratorKey(PackageGuid);
+            if (generatorKey == null)
+            {
+                return VSConstants.E_FAIL;
+            }
+
+            if (!generatorKey.GetSubKeyNames().Contains(wszProgId))
+            {
+                return VSConstants.E_FAIL;
+            }
+
+            var progKey = generatorKey.OpenSubKey(wszProgId, false);
+
+            // The clsid value is the only required value. The other 3 are optional
+            if (!progKey.GetValueNames().Contains(CLSIDKey))
+            {
+                return VSConstants.E_FAIL;
+            }
+
+            pguidGenerator = Guid.Parse(progKey.GetValue<string>(CLSIDKey));
+
+            // Explicitly convert anything that's not 1 to 0
+            pbGeneratesDesignTimeSource = progKey.GetValue<int>(DesignTimeSourceKey, 0) == 1 ? 1 : 0;
+            pbGeneratesSharedDesignTimeSource = progKey.GetValue<int>(SharedDesignTimeSourceKey, 0) == 1 ? 1 : 0;
+            pbUseTempPEFlag = progKey.GetValue<int>(DesignTimeCompilationFlagKey, 0) == 1 ? 1 : 0;
+
+            return VSConstants.S_OK;
+        }
+
+        protected override void Dispose(bool disposing)
+        {
+            if (disposing)
+            {
+                _settingsRoot.Dispose();
+                _settingsRoot = null;
+            }
+        }
+
+        protected override void Initialize()
+        {
+            _settingsRoot = _registryHelper.RegistryRoot(_serviceProvider, __VsLocalRegistryType.RegType_Configuration, false);
```

Comment:
We shouldn't leaving the handle open for the life of the project. Registry key handles should be opened for a short period of time.